### PR TITLE
Clarify what Lotus snapshots actually do.

### PR DESCRIPTION
**Pull request by johnnymatthews**
There's a bit of confusion surrounding what Lotus snapshots actually do. This PR:

- [x] Explains the difference between Lightweight and Full-chain snapshots.
- [x] Walks the reader through how to use them.
- [x] Points out who shouldn't use these snapshots.

### CODE REVIEW

**Comment by johnnymatthews** [comment]
This PR now also cleans up the rest of the info about snapshots.

File: docs/get-started/lotus/chain.md [hunk 1]
```diff
@@ -17,113 +15,150 @@ Lotus will automatically sync to the latest _chain head_ by:
...
+| [Full chain](#full-chain-snapshot)   | asd         | Current block | Genesis block               | Current block - 2000 blocks |
 
-We recommend most users to perform the initial node sync from a minimal, lightweight snapshot. Trusted state snapshots do not contain the full chain and might not be suitable for nodes that need to perform queries against historical state information, like block explorers, but otherwise work for most users.
+### Lightweight snapshot
 
-A recent minimal trusted state chain snapshot is available [here](https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_latest.car). We can instruct lotus to start the daemon and directly import from the URL:
+We recommend most users perform the initial node sync from a lightweight snapshot. These snapshots do not contain the full chain and are not suitable for nodes that need to perform queries against historical state information, such as block explorers. However, they are significantly smaller than full chain snapshots and should be sufficient for most use-cases.
```
Comment (jennijuju) [inline docs/get-started/lotus/chain.md:29]:
```suggestion
We recommend most users perform the initial node sync from a lightweight snapshot. These snapshots do not contain the full states of the chain and are not suitable for nodes that need to perform queries against historical state information, such as block explorers. However, they are significantly smaller than full chain snapshots and should be sufficient for most use-cases.
```

File: docs/get-started/lotus/chain.md [hunk 2]
```diff
@@ -17,113 +15,150 @@ Lotus will automatically sync to the latest _chain head_ by:
...
 
-```sh
-# The snapshot size is about 7GiB. This works for mainnet.
-lotus daemon --import-snapshot https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_latest.car
+:::warning
+These lightweight state snapshots **do not contain any message receipts**. To get message receipts, you need to sync your Lotus node from the genesis block without using any of these snapshots.
```
Comment (jennijuju) [inline docs/get-started/lotus/chain.md:32]: I believe, if you have synced the chain and have the states, then you can `set-head` to a previous epoch and resync from the peers to get EVERYTHING, including receipts. where the epoch one set-head to must be a block they have the state for. @ribasushi could you please confirm?

**Review by jennijuju — DISMISSED** [review]

File: docs/get-started/lotus/chain.md (hunk 2 above)
Comment (ribasushi) [reply docs/get-started/lotus/chain.md:32]: @jennijuju not exactly. By default a `set-head` to something you do not already have in full locally will fail. You'd need to set the underdocumented `LOTUS_ENABLE_CHAINSTORE_FALLBACK=1`

Comment (johnnymatthews) [reply docs/get-started/lotus/chain.md:32]: Just to make sure I understand you both correctly, in order for a user to get all the message receipts on their now after using a snapshot they need it:

- Enable `LOTUS_ENABLE_CHAINSTORE_FALLBACK=1`
- Use `set-head`.
- Wait for a very long time.

Is there any benefit to doing this, or would it be as fast/faster to just sync a node normally (without using a snapshot)?

Comment (ribasushi) [reply docs/get-started/lotus/chain.md:32]: There is still a benefit - on each epoch transition you do not wait to get the next set of messages via libp2p. All your messages for "next tick" are already magically there: you just imported them as part of the 150+GiB (and growing ) snapshot.

Unfortunately "long time" and "slightly less long time" is pretty damn long either way ( ~13 days currently, given screaming hardware )

Comment (johnnymatthews) [reply docs/get-started/lotus/chain.md:32]: Ok, that makes sense. I'll add these steps in.

File: docs/get-started/lotus/chain.md [hunk 3]
```diff
@@ -17,113 +15,163 @@ Lotus will automatically sync to the latest _chain head_ by:
...
+1. Import the lightweight snapshot [as normal](#lightweight-snapshot).
+1. Create a new environment variable called `LOTUS_ENABLE_CHAINSTORE_FALLBACK` and set it to `1`:
 
-### Syncing from a full snapshot (mainnet)
+   ```profile
+   $LOTUS_ENABLE_CHAINSTORE_FALLBACK=1
+   ```
 
-An alternative to trusted state snapshots are full chain snapshots ([available here](https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/complete_chain_with_finality_stateroots_latest.car)), which are significantly bigger and therefore slower to download, but will contain every block from genesis until the current tipset. In addition to the trusted import option described above, these complete snapshots can be trustlessly imported by fully recalculating the state during import. Simply supply the `--import-chain` option instead:
+1. <!-- USE LOTUS SET-HEAD TO DO SOMETHING -->
```
Comment (johnnymatthews) [inline docs/get-started/lotus/chain.md:70]: @ribasushi @jennijuju could either of your expand on the actual steps needed here? I'm completely guessing, and don't have a miner/node available to test.

File: docs/get-started/lotus/chain.md [hunk 4]
```diff
@@ -50,12 +50,25 @@ These lightweight state snapshots **do not contain any message receipts**. To ge
...
     ```
 
-You can skip the `sha256sum` check and use the snapshot URL directly if you'd prefer:
+We strongly recommend you verify the checksum of the download. However, you can skip the `sha256sum` check and use the snapshot URL directly if you'd prefer:
 
 ```bash
 lotus daemon --import-snapshot https://fil-chain-snapshots-fallback.s3.amazonaws.com/mainnet/minimal_finality_stateroots_latest.car
 ```
 
+#### Get message receipts
```
Comment (ribasushi) [inline docs/get-started/lotus/chain.md:59]: I think this is getting things more confusing overall. My suggestion is to not mention receipts here at all.

Comment (ribasushi) [reply docs/get-started/lotus/chain.md:59]: Sorry, wasn't clear enough. Receipts are not something that is included in snapshots, period: https://github.com/filecoin-project/lotus/issues/3849

What I am saying is: let's not talk about receipts *at all*. They are an advanced thing that a very small part of the population care about...

Comment (johnnymatthews) [reply docs/get-started/lotus/chain.md:59]: Ok, follow up questions: 

1. Both snapshots contain messages, but neither snapshot contains message receipts. Correct?
2. The best and easiest way to get message receipts is **not** use any snapshots and to just sync the chain normally like a pleb. Correct?

**Pull request merged**